tests/eas/cpuhp.py

```diff
@@ -109,14 +94,22 @@ def _experimentsInit(cls, *args, **kwargs):
         cls._log.info('Hotpluggable CPUs found on target: ')
         cls._log.info(cls.hotpluggable_cpus)
 
+        # Mark the target as alive for now
+        cls.target_alive = True
+
         # Run random hotplug sequence on target
         cls.cpuhp_seq_script = cls._random_cpuhp_script()
         cls.cpuhp_seq_script.push()
         cls.cpuhp_seq_script.run(as_root=True, background=True)
```

Comment:
Devlib's `execute` has a `timeout` param. What about extending `target_script` to forward this param (it only makes sense for `background=False`, but that's fine)?

Not sure if it's **exactly** the same in terms of end results to launching the script in background and then checking for responsiveness after some amount time though...

Reply:
So, as of now, the script on the target is pretty stupid and runs in an infinite loop. So the host has to intervene to stop it, we can't just wait for it to finish or throw an exception ... The reason I originally implemented it that way is because bash on android is 'funny' so I just went for the simplest possible thing to avoid coping with that ...

Reply:
Ok I won't question that :)

Reply:
Actually maybe I can do something by using `timeout` http://man7.org/linux/man-pages/man1/timeout.1.html
I'll look into it